src/test/unit/test.js

```diff
@@ -491,6 +491,51 @@ describe('Client', function() {
     })
   })
```

Comment:
Would you like me to change the style of the tests as well? One issue with functional tests *instead* of unit tests in this case is we couldn't ensure that the outgoing request has the same payload each time — we'd only be testing that the server interpreted it correctly, as opposed to testing whether or not we sent the right payload.